ansible-collection-microsoft-sql/ansible-collection-microsoft-sql.spec

```diff
@@ -54,159 +35,50 @@ BuildRequires: ansible >= 2.9.10
 %endif
 
 %if %{without ansible}
-# Empty command. We don't have ansible-galaxy.
-%define ansible_collection_build() tar -cf %{_tmppath}/%{collection_namespace}-%{collection_name}-%{version}.tar.gz .
-%else
-%define ansible_collection_build() ansible-galaxy collection build
-%endif
-
-%if %{without ansible}
+# We don't have ansible-galaxy.
 # Simply copy everything instead of galaxy-installing the built artifact.
-%define ansible_collection_install() mkdir -p %{buildroot}%{ansible_collection_files}/%{collection_name}; (cd %{buildroot}%{ansible_collection_files}/%{collection_name}; tar -xf %{_tmppath}/%{collection_namespace}-%{collection_name}-%{version}.tar.gz)
+%define ansible_collection_build_install() tar -cf %{_tmppath}/%{collection_namespace}-%{collection_name}-%{version}.tar.gz .; mkdir -p %{buildroot}%{ansible_collection_files}/%{collection_name}; (cd %{buildroot}%{ansible_collection_files}/%{collection_name}; tar -xf %{_tmppath}/%{collection_namespace}-%{collection_name}-%{version}.tar.gz)
 %else
-%define ansible_collection_install() ansible-galaxy collection install -n -p %{buildroot}%{_datadir}/ansible/collections %{collection_namespace}-%{collection_name}-%{version}.tar.gz
+%define ansible_collection_build_install() ansible-galaxy collection build; ansible-galaxy collection install -n -p %{buildroot}%{_datadir}/ansible/collections %{collection_namespace}-%{collection_name}-%{version}.tar.gz
 %endif
 
-# For each role, call either defcommit() or deftag(). The other macros
-# (%%id and %%shortid) can be then used in the same way in both cases.
-# This way  the rest of the spec file des not need to know whether we are
-# dealing with a tag or a commit.
+# For each role, call defcommit() and the point to it with SourceN: %{archiveurlN}.
 %global archiveext tar.gz
-# list of role names
+# list of source role names
 %global rolenames %nil
 # list of assignments that can be used to populate a bash associative array variable
 %global rolestodir %nil
+# list of target rolenames to copy the roles to
+%global target_rolenames %nil
```

Comment:
can't you use `%rolenameN` for the target name and `%repo` for the original name? Not sure whether this will work, but some kind of renaming is already done for the sshd role this way (the repo is called `ansible-sshd` and we call the role `linux-system-roles.sshd`).

Reply:
Hi Pavel, thank you for the review. Indeed I now see that the original linux-system-roles spec file had a solution to this and I overviewed it.
I have tried to implement this and faced an issue. To rename the role during the conversion to the collection, the original role must have a corresponding name. After I used the solution you proposed, rpmbuild extracts the `mssql` role and copies it to the `sql-server` directory. Then, it runs the role2collection script like that (pasting the parsed command from the buildlog for readability):
```
python3 lsr_role2collection.py --role sql-server \
--src-path sql-server \
--dest-path .collections \
--readme lsr_role2collection/collection_readme.md \
--namespace microsoft \
--collection sql \
--new-role server
```
Because the `--new-role` is set to `server`, the script tries to rename the role from `sql-server` to `server`, however, the role in fact uses the `mssql` wording everywhere. So the issue is that the scrip tries to rename e.g. `linux-system-roles.sql-server` to `microsoft.sql.server`, which is incorrect because in fact the role is called as `linux-system-roles.mssql`.

I tried to use the `--extra-mapping` option that @nhosoi added to lsr_role2collection.py instead of `--new-role` like `--extra-mapping mssql:server` but it didn't work for me. I suppose this option does something else. Noriko, can you justify?

Reply:
Apparently, to run this lsr_role2collection.py script the spec file will need to declare an array containing the source name and the target name, exactly what `TARGET_NAMES` does. So after all I think there is no need to re-use the `repo` macro, rather get rid of it as I have `target_namesN` for that. And after all mssql is not only the repo name but the actually name of the source role.

Reply:
@pcahyna so using the existing `repo` macro would require the same functionality as the `target_rolenames` macro does, and I find the `target_rolename` macro names more descriptive than `repo`, so I kept `target_rolename` and got rid of unused `repo` and `forge` macros. Does this sound fine to you?

Reply:
well `repo` is not unused, it is used to construct the complete URL (it is the last part of the complete URL of the repo). Sure, you can remove it and construct the URL in a different way, but then your spec file will become a bit different from what is used in linux-system-roles. Since linux-system-roles could also use this kind of renaming mechanics, I think it would be better to share it, but at the end it is up to you.

Reply:
I think I am not going to keep the two spec files as identical as possible. They are too different. The reasoning behind keeping them identical and e.g. keeping unused macros would be to enable easy copying of changes from linux-system-roles.spec to ansible-collection-microsoft-sql.spec. I do not think that this will be the case because the files are already much different and are used differently. So picking changes from linux-system-roles anyway will be a manual process of reviewing each change and decing if it is needed or not.
So I think here I am trying to create a new spec file basing on the linux-system-roles.spec. This spec file will always be smaller and simpler.

Reply:
ok - but if you figure out a better way to do something in linux-system-roles.spec, please submit a PR

Reply:
@spetrosi, sorry for my slow response. I think you already reached the conclusion, but just in case...

> I tried to use the --extra-mapping option that @nhosoi added to lsr_role2collection.py instead of --new-role like --extra-mapping mssql:server but it didn't work for me. I suppose this option does something else. Noriko, can you justify?

That's correct. --extra-mapping does not step on toes of --new-role to avoid the ambiguity.